lib/model.js

```diff
@@ -644,20 +644,24 @@ class Model extends Entity {
             return null;
         }
 
-        const isJoiSchema = !is.undef(schema._joi);
+        const isJoiSchema = schema.isJoi;
 
         let sanitized;
         let joiOptions;
         if (isJoiSchema) {
-            sanitized = schema._joi.validate(data).value;
+            const { error, value } = schema.validateJoi(data);
+            if (!error) {
+                sanitized = Object.assign({}, value);
+            }
             joiOptions = schema.options.joi.options || {};
-        } else {
+        }
+        if (sanitized === undefined) {
             sanitized = Object.assign({}, data);
         }
 
         const isSchemaExplicitOnly = isJoiSchema
-            ? !joiOptions.allowUnknown
-            : schema.options.explicitOnly !== false;
+            ? joiOptions.stripUnknown || !joiOptions.allowUnknown
```

Comment:
Initially I thought it was logical to set `isSchemaExplicitOnly` to `true` when `strupUnknown: false` and `allowUnknown: false` since `allowUnknown` implies the user wants unknown keys stripped. However, `stripUnknown` is already defaulted to `allowUnknown !== true` so this "extra" default would only be used when the user explicitly sets `allowUnkown: false` and `stripUnknown: false`

Reply:
I see, thanks for the explanation.